floors/enforce.go

```diff
@@ -0,0 +1,251 @@
+package floors
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/currency"
+	"github.com/prebid/prebid-server/exchange/entities"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// EnforceFloors does floors enforcement for bids from all bidders based on floors provided in request, account level floors config
+func Enforce(bidRequestWrapper *openrtb_ext.RequestWrapper, seatBids map[openrtb_ext.BidderName]*entities.PbsOrtbSeatBid, account config.Account, conversions currency.Conversions) (map[openrtb_ext.BidderName]*entities.PbsOrtbSeatBid, []error, []*entities.PbsOrtbSeatBid) {
+	rejectionErrs := []error{}
+
+	rejectedBids := []*entities.PbsOrtbSeatBid{}
+
+	requestExt, err := bidRequestWrapper.GetRequestExt()
+	if err != nil {
+		return seatBids, []error{errors.New("Error in getting request extension")}, rejectedBids
+	}
+
+	if !isPriceFloorsEnabled(account, bidRequestWrapper) {
+		return seatBids, []error{errors.New("Floors feature is disabled at account or in the request")}, rejectedBids
+	}
+
+	if isSignalingSkipped(requestExt) || !isValidImpBidFloorPresent(bidRequestWrapper.BidRequest.Imp) {
+		return seatBids, nil, rejectedBids
+	}
+
+	enforceFloors := isSatisfiedByEnforceRate(requestExt, account.PriceFloors.EnforceFloorsRate, rand.Intn)
```

Comment:
`enforceFloors` is updated into` req.ext.prebid.floors.enforcement.enforcePBS` using `updateEnforcePBS()`,  and function `enforceFloorToBids()` checks for same flag before enforcement. 

`enforceFloorToBids()` function along with floors enforcement it also updates bid extension for floors details in case signalling is enabled and enforcement is disabled. This done to avoid one more iterative loop to iterate bids in separate
 function to update bid extension for floors related details. 
 
 We can also make two function 
 `
 updateBidExt()
 if enforceFloors {
         enforceFloorToBids()
 }
 `
 thoughts ?

Reply:
Ah yes I do think this is better. During my initial review I was thinking that it might be better from a readability and code organization perspective to separate the responsibilities in `enforceFloorToBids` so that you perform all of your bid ext updates in one place and determine which bids are eligible/rejected in another. I figured though that you landed on your current implementation so you didn't have to iterate over the bids twice. My take is that there shouldn't be any performance penalty by iterating over the bids twice given that the number of items you would iterate over is very small so two loops should be fine. In that case, I would advocate for making a change but I'll leave that up to you.